### PR TITLE
Add support for opening file:// urls

### LINKED ISSUE #2304: file:// URLs not opening with Cmd-U
## Expected Behavior

If an entry contains a URL of the sort file://, then Cmd-U should open the local file with the default program registered with Mac OS for that particular file type.

## Current Behavior

Nothing seems to happen.

## Possible Solution

Sorry, but I don't have the slightest idea. This stopped working when I moved from my old Macbook Pro to a new one including an upgrade from Mac OS Sierra to High Sierra.

## Steps to Reproduce (for bugs)

1. Get a Mac with High Sierra installed.
2. Install KeePassXC 2.3.4
3. Create an entry like this: Title: Test, URL: file:///Users/pmh/Desktop/Test.txt - replace "pmh" with your user name. Create a text file named "Test.txt" on your desktop.
4. Highlight the entry, type Cmd-U

## Context

This is important for our team workflow. We have a central Nextcloud shared repository of .kdbx files. The passwords to these files every operator puts into his own personal keepass. To open a particular file, the established way is:
- unlock personal keepass
- select file
- Cmd-C, Cmd-U, Cmd-V, Enter

## Debug Info

KeePassXC - Version 2.3.4
Revision: 6fe821c

Bibliotheken:
- Qt 5.9.3
- libgcrypt 1.8.2

Betriebssystem: macOS High Sierra (10.13)
CPU-Architektur: x86_64
Kernel: darwin 17.7.0

Aktivierte Erweiterungen:
- Auto-Type
- Browser Integration
- Legacy Browser Integration (KeePassHTTP)
- SSH Agent
- YubiKey

### PR DESCRIPTION
**Pull request by weslly**
## Motivation and context
Fix #2304.

## How has this been tested?
I added a couple of test cases for both Unix and Windows file paths.

## Types of changes
- ✅ Bug fix (non-breaking change which fixes an issue)

## Checklist:
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have added tests to cover my changes.

### CODE REVIEW

File: tests/TestEntry.cpp [hunk 1]
```diff
@@ -156,6 +158,9 @@ void TestEntry::testResolveUrl()
     QCOMPARE(entry->resolveUrl(""), QString(""));
     QCOMPARE(entry->resolveUrl(testUrl), "https://" + testUrl);
     QCOMPARE(entry->resolveUrl("http://" + testUrl), "http://" + testUrl);
+    // Test file:// URL's
+    QCOMPARE(entry->resolveUrl("file://" + testFileUnix), "file://" + testFileUnix);
+    QCOMPARE(entry->resolveUrl("file:///" + testFileWindows), "file:///" + testFileWindows);
```
Comment (droidmonkey) [inline tests/TestEntry.cpp]: Here is my issue with this. If you didn't add the "file:///" at this function call, then the function "resolveUrl" would prepend an "https://" to the file location (ie, "https://c:/WINDOWS/test.txt") which is obviously not desired. I think we need to refactor the resolveUrl function a little.

Comment (weslly) [reply tests/TestEntry.cpp]: You mean prepending the schema at `resolveUrl()` if the url starts with a slash or `[a-Z]:/`?

Comment (droidmonkey) [reply tests/TestEntry.cpp]: Yes, that would be an acceptable option. The regex would be ```([a-Z]:)?\/```

Comment (weslly) [reply tests/TestEntry.cpp]: @droidmonkey fixed

**Pull request merged**